File: app/src/main/java/com/example/javatokotlin/java/EventCallback.java
package com.example.javatokotlin.java;

import android.os.Handler;
import android.util.Log;

/**
 * EventCallback
 * Created by pdc-k-kamiya on 2018/03/17.
 */
public class EventCallback {

    public interface Callback{
        void onCall();
    }

    private Callback mCallback;

    public EventCallback(Callback callback) {
        mCallback = callback;
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                mCallback.onCall();
            }
        }, 1000);
    }
}

class EventControl{
    public EventControl() {
        new EventCallback(new EventCallback.Callback() {
            @Override
            public void onCall() {
                Log.d("Callback", "Callback");
            }
        });
    }
}

File: app/src/main/java/com/example/javatokotlin/kotlin/EventCallback.kt
package com.example.javatokotlin.kotlin

import android.os.Handler
import android.util.Log

/**
 * EventCallback
 * Created by pdc-k-kamiya on 2018/03/17.
 */
class EventCallback(callback:(() -> Unit)){
    init {
        // 1秒後に実行
        Handler().postDelayed({ callback.invoke() }, 1000)
    }
}

class EventControl{
    init {
        EventCallback(callback = { Log.d("Callback", "Callback") })
    }
}